ntoskrnl/io/pnpmgr/devaction.c

```diff
@@ -957,25 +931,19 @@ NTSTATUS
 IopQueryHardwareIds(PDEVICE_NODE DeviceNode,
                     HANDLE InstanceKey)
 {
-    IO_STACK_LOCATION Stack;
-    IO_STATUS_BLOCK IoStatusBlock;
+    PWCHAR HwId;
```

Comment:
Use `PWSTR` if the pointer will point to a string that is NULL-terminated.
Also, if the string actually points to a (null-termianted) **multi-string**, then the `PZZWSTR` type may be better (and the name be: `HwIds` in plural).